server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -210,6 +210,10 @@ public final void run() {
                     throw new SearchPhaseExecutionException(getName(), msg, null, ShardSearchFailure.EMPTY_ARRAY);
                 }
             }
+            if (checkMinimumVersion(shardsIts) == false) {
```

Comment:
You don't need to check the minimum version if it's equal to the minimum compatible version (default value).